proj/src/main/webapp/script.js

```diff
@@ -26,17 +27,26 @@ function initRef() {
 function pushChatMessage() {
     const messageRef = firebase.database().ref(path);
 
-    const message = document.getElementById('message-input').value;
+    const messageInput = document.getElementById('message-input');
     // push message to datastore
-    messageRef.push(message);
-} 
+    messageRef.push(messageInput.value);
+    messageInput.value = null; // clear the message
+}
+
+/**
+    so pressing enter instead of using the button submits the form,
+    which is not what we want because it reloads the page. so this
+    makes it so when they press enter, it presses the send button
+ */
+function clickWithEnterKey() {
+    const messageInput = document.getElementById('message-input');
 
-// Your web app's Firebase configuration
-var firebaseConfig = {
-    
-};
-// Initialize Firebase
-firebase.initializeApp(firebaseConfig);
+    messageInput.addEventListener('keyup', function(event) {
+        if (event.keyCode === 13) { // 13 is the keycode for the enter key
+            document.getElementById('send-button').click();
```

Comment:
So you're setting up a chain of actions here that trigger each other: press enter → click "send" button →call `pushChatMessage()`.  I think it would be better to shortcut the chain and just make it this instead: press enter → call `pushChatMessage()`.  Does that make sense?

Reply:
I agree with showermat, this can just call `pushChatMessage()`

Reply:
yeah that makes sense. fixed